usr/sbin/registercloudguest

```diff
@@ -377,32 +376,30 @@ if not os.path.exists(utils.get_state_dir()):
 # Write the data of the current target server
 utils.set_as_current_smt(registration_target)
 
-# Check if we need to send along any instance data
-instance_data_filepath = utils.get_state_dir() + str(uuid.uuid4())
-instance_data = utils.get_instance_data(cfg)
-if instance_data:
-    inst_data_out = open(instance_data_filepath, 'w')
-    inst_data_out.write(instance_data)
-    inst_data_out.close()
-
 # Check if registration is supported
 if not utils.is_registration_supported(cfg):
     sys.exit(0)
 
+# Check if we need to send along any instance data
+utils.get_instance_data(cfg)
+
```

Comment:
For a future improvement, it might be good to clear the cache with a context manager, to reduce the risk of forgetting to include the `clear_instance_data_cache` before each `exit`